packages/bolt-connection/test/lang/functional.test.js

```diff
@@ -0,0 +1,170 @@
+import { reuseOngoingRequest } from '../../src/lang/functional.js'
+
+describe('functional', () => {
+  describe('reuseOnGoingRequest', () => {
+    it('should call supplied function with the params', async () => {
+      const expectedParams = ['a', 1, { a: 'a' }]
+      const func = jest.fn(() => Promise.resolve())
+
+      const decoratedFunction = reuseOngoingRequest(func)
+      await decoratedFunction(...expectedParams)
+
+      expect(func).toHaveBeenCalledWith(...expectedParams)
+    })
+
+    it('should call supplied function with this', async () => {
+      const expectedParams = ['a', 1, { a: 'a' }]
+      const thisArg = { t: 'his' }
+      const func = jest.fn(async function () {
+        return thisArg
+      })
+
+      const decoratedFunction = reuseOngoingRequest(func, thisArg)
+      const receivedThis = await decoratedFunction(...expectedParams)
+
+      expect(receivedThis).toBe(thisArg)
+    })
+
+    it('should values return by the supplied function', async () => {
+      const expectedResult = { a: 'abc' }
+      const func = jest.fn(() => Promise.resolve(expectedResult))
+
+      const decoratedFunction = reuseOngoingRequest(func)
+      const result = await decoratedFunction()
+
+      expect(result).toBe(expectedResult)
+    })
+
+    it('should throw value thrown by supplied function', async () => {
+      const error = new Error('Ops, I did it!')
```

Comment:
```suggestion
      const error = new Error('Oops, I did it again!')
```
https://www.youtube.com/shorts/PydX9fnaZ3U

:rofl: